website/server/models/group.js

```diff
@@ -542,8 +612,8 @@ function setUserStyles (newMessage, user) {
   newMessage.markModified('userStyles');
 }
 
-schema.methods.sendChat = function sendChat (message, user, metaData) {
-  let newMessage = chatDefaults(message, user);
+schema.methods.sendChat = function sendChat (message, user, metaData, info = {}) {
```

Comment:
the signature of this function should be changed: instead of having 4 paraneters and passing `null, null` a lot of times it should be changed to accept a single object parameter:

```
schema.methods.sendChat = function sendChat (options = {}) {
    const {message, user, metadata, info = {}} = options
    ....
}
```

Reply:
OK

Reply:
I was about to implement those suggestions today, but this change in particular breaks a lot of tests, and requires changing code in a lot of scripts.
Maybe I complete it by this weekend...